tests/adapters/unified_tests/vlan_management_test.py

```diff
@@ -21,168 +24,178 @@ class VlanManagementTest(ConfiguredTestCase):
+        self.client.add_vlan(number=2999, name="my-test-vlan")
+        with self.assertRaises(UnknownResource): # todo different exceptions ...
+            self.client.remove_trunk_vlan(interface_id=self.test_port, vlan=2999)
+        self.client.remove_vlan(number=2999)
 
-        self.put("/switches/{switch}/interfaces/{port}/port-mode", raw_data='access')
+        self.client.set_access_mode(interface_id=self.test_port)
 
     @skip_on_switches("juniper", "juniper_qfx_copper")
     def test_invalid_interface_parameter_fails(self):
-        response = self.put("/switches/{switch}/interfaces/42/9999/shutdown", raw_data='true', fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        with self.assertRaises(UnknownInterface):
+            self.client.openup_interface('42/9999')
 
-        response = self.put("/switches/{switch}/interfaces/42/9999/shutdown", raw_data='false', fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        with self.assertRaises(UnknownInterface):
+            self.client.shutdown_interface('42/9999')
 
-        response = self.put("/switches/{switch}/interfaces/42/9999/port-mode", raw_data='access', fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        with self.assertRaises(UnknownInterface):
+            self.client.set_access_mode('42/9999')
 
-        response = self.put("/switches/{switch}/interfaces/42/9999/port-mode", raw_data='trunk', fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        with self.assertRaises(UnknownInterface):
+            self.client.set_trunk_mode('42/9999')
 
-        response = self.delete("/switches/{switch}/interfaces/42/9999/access-vlan", fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        with self.assertRaises(UnknownResource): # todo multiple exceptions
+            self.client.set_access_vlan('42/9999', 1234)
 
-        response = self.delete("/switches/{switch}/interfaces/42/9999/access-vlan", fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        with self.assertRaises(UnknownInterface):
+            self.client.remove_access_vlan('42/9999')
 
-        response = self.post("/switches/{switch}/interfaces/42/9999/trunk-vlans", data=2999, fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        # todo Dell 10G raises UnknownInterface
+        # todo other raises UnknownVlan
+        with self.assertRaises(UnknownResource):
+            self.client.add_trunk_vlan('42/9999', 2999)
 
-        self.delete("/switches/{switch}/interfaces/{port}/trunk-native-vlan", fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404))
+        # todo Brocade raises UnknownInterface
+        # todo Dell 10G raises NativeVlanNotSet
+        # todo Cisco does not raise???
+        if self.switch_type != 'cisco':
```

Comment:
No if typing in the test, if Cisco doesn't comply we either fix Cisco or add it to the skip list annotation